src/catalog/catalog.cpp

```diff
@@ -30,421 +30,714 @@ Catalog *Catalog::GetInstance(void) {
+  IndexCatalog::GetInstance()->InsertIndex(
+      COLUMN_CATALOG_PKEY_OID, COLUMN_CATALOG_NAME "_pkey", COLUMN_CATALOG_OID,
+      IndexType::BWTREE, IndexConstraintType::PRIMARY_KEY, true,
+      std::vector<oid_t>({0, 1}), pool_.get(), nullptr);
+  IndexCatalog::GetInstance()->InsertIndex(
+      COLUMN_CATALOG_SKEY0_OID, COLUMN_CATALOG_NAME "_skey0",
+      COLUMN_CATALOG_OID, IndexType::BWTREE, IndexConstraintType::UNIQUE, true,
+      std::vector<oid_t>({0, 2}), pool_.get(), nullptr);
+  IndexCatalog::GetInstance()->InsertIndex(
+      COLUMN_CATALOG_SKEY1_OID, COLUMN_CATALOG_NAME "_skey1",
+      COLUMN_CATALOG_OID, IndexType::BWTREE, IndexConstraintType::DEFAULT,
+      false, std::vector<oid_t>({0}), pool_.get(), nullptr);
+
+  IndexCatalog::GetInstance()->InsertIndex(
+      INDEX_CATALOG_PKEY_OID, INDEX_CATALOG_NAME "_pkey", INDEX_CATALOG_OID,
+      IndexType::BWTREE, IndexConstraintType::PRIMARY_KEY, true,
+      std::vector<oid_t>({0}), pool_.get(), nullptr);
+  IndexCatalog::GetInstance()->InsertIndex(
+      INDEX_CATALOG_SKEY0_OID, INDEX_CATALOG_NAME "_skey0", INDEX_CATALOG_OID,
+      IndexType::BWTREE, IndexConstraintType::UNIQUE, true,
+      std::vector<oid_t>({1}), pool_.get(), nullptr);
+  IndexCatalog::GetInstance()->InsertIndex(
+      INDEX_CATALOG_SKEY1_OID, INDEX_CATALOG_NAME "_skey1", INDEX_CATALOG_OID,
+      IndexType::BWTREE, IndexConstraintType::DEFAULT, false,
+      std::vector<oid_t>({2}), pool_.get(), nullptr);
+
+  // Insert pg_catalog database into pg_database
+  pg_database->InsertDatabase(CATALOG_DATABASE_OID, CATALOG_DATABASE_NAME,
+                              pool_.get(), nullptr);
+
+  // Insert catalog tables into pg_table
+  pg_table->InsertTable(DATABASE_CATALOG_OID, DATABASE_CATALOG_NAME,
+                        CATALOG_DATABASE_OID, pool_.get(), nullptr);
+  pg_table->InsertTable(TABLE_CATALOG_OID, TABLE_CATALOG_NAME,
+                        CATALOG_DATABASE_OID, pool_.get(), nullptr);
+  pg_table->InsertTable(INDEX_CATALOG_OID, INDEX_CATALOG_NAME,
+                        CATALOG_DATABASE_OID, pool_.get(), nullptr);
+  pg_table->InsertTable(COLUMN_CATALOG_OID, COLUMN_CATALOG_NAME,
+                        CATALOG_DATABASE_OID, pool_.get(), nullptr);
+}
 
-  // Create table for database metrics
-  auto database_metrics_catalog = CreateMetricsCatalog(default_db_oid,
-      DATABASE_METRIC_NAME);
-  default_db->AddTable(database_metrics_catalog.release(), true);
+//===----------------------------------------------------------------------===//
+// CREATE FUNCTIONS
+//===----------------------------------------------------------------------===//
 
-  // Create table for index metrics
-  auto index_metrics_catalog = CreateMetricsCatalog(default_db_oid,
-      INDEX_METRIC_NAME);
-  default_db->AddTable(index_metrics_catalog.release(), true);
+ResultType Catalog::CreateDatabase(const std::string &database_name,
+                                   concurrency::Transaction *txn) {
```

Comment:
Yeach, we are currently working on that. Since binder needs to read record from catalog tables, dose that mean transaction should begin in binder instead of executor??